Sources/PackageDescription/PackageDescription.swift

```diff
@@ -278,6 +279,50 @@ public final class Package {
         self.cxxLanguageStandard = cxxLanguageStandard
         registerExitHandler()
     }
+    
+    /// Initializes a Swift package with configuration options you provide.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the Swift package, or `nil` to use the package's Git URL to deduce the name.
+    ///   - defaultLocalization: The default localization for resources.
+    ///   - platforms: The list of supported platforms with a custom deployment target.
+    ///   - pkgConfig: The name to use for C modules. If present, Swift Package Manager searches for a
+    ///   `<name>.pc` file to get the additional flags required for a system target.
+    ///   - providers: The package providers for a system target.
+    ///   - products: The list of products that this package makes available for clients to use.
+    ///   - dependencies: The list of package dependencies.
+    ///   - targets: The list of targets that are part of this package.
+    ///   - swiftLanguageModes: The list of Swift language modes with which this package is compatible.
+    ///   - cLanguageStandard: The C language standard to use for all C targets in this package.
+    ///   - cxxLanguageStandard: The C++ language standard to use for all C++ targets in this package.
+    @available(_PackageDescription, introduced: 6)
+    public init(
```

Comment:
How is this not ambiguous for scenarios when `swiftLanguageModes` are not provided? It should be similar to i.e.:
```
func test(x: Int, b: Int = 0) {}
func test(x: Int, c: Int = 1) {}

test(x: 42)
```

where `test(x: 42)` is an ambiguous reference.

Reply:
The previous init method is marked as obsoleted in 6. So there is only one valid init method.

This appears to be the approach that has been used for all the previous init methods.

Reply:
“Obsoleted” doesn’t add anything in terms of weight for overload ranking so both methods are equally valid in swift 6 mode.

Reply:
Oh I confused it with “deprecated”, sorry!